# HR ATTRITION ANALYSIS

**by Jayesh Suryawanshi**  

**Business Task:**  

To peform attrition analysis using provided information (consisting of survey results) to obtain insights.

**Information on the Data Source:**  
The data source consists of an excel file called **HR DATA** which consists of the demographic information about employees such as their age, gender, education and other information such as their job role, department, wages, performance rating work hours, training time required, etc. Alongside these, there is also survey data providing information on their job, relationship, environment satisfaction, work life balance, etc.

Tools Used: PostgreSQL (Data Analysis), Tableau (Data Visualization)


In [1]:
%load_ext sql

In [2]:
import os

In [3]:
host = "localhost"
database = "hr_attrition"
user = 'user_name'
password = 'password'

In [4]:
connection_string = f"postgresql://{user}:{password}@{host}/{database}"

In [5]:
%sql $connection_string

In [8]:
%%sql

select * from hr_data limit 10;

 * postgresql://postgres:***@localhost/hr_attrition
10 rows affected.


[('Yes', 'Travel_Rarely', '35 - 44', 'Ex-Employees', 'Sales', 'Life Sciences', 'STAFF-1', 1, 'Female', 'Sales Executive', 'Single', 'Yes', 'Y', 0, 41, 0, 1102, 1, 'Associates Degree', 1, 2, 94, 3, 2, 4, 5993, 19479, 8, 11, 3, 1, 80, 0, 8, 1, 6, 4, 0, 5),
 ('No', 'Travel_Frequently', '45 - 54', 'Current Employees', 'R&D', 'Life Sciences', 'STAFF-2', 2, 'Male', 'Research Scientist', 'Married', 'No', 'Y', 3, 49, 1, 279, 8, 'High School', 1, 3, 61, 2, 2, 2, 5130, 24907, 1, 23, 4, 4, 80, 1, 10, 3, 10, 7, 1, 7),
 ('Yes', 'Travel_Rarely', '35 - 44', 'Ex-Employees', 'R&D', 'Other', 'STAFF-4', 4, 'Male', 'Laboratory Technician', 'Single', 'Yes', 'Y', 3, 37, 0, 1373, 2, 'Associates Degree', 1, 4, 92, 2, 1, 3, 2090, 2396, 6, 15, 3, 2, 80, 0, 7, 3, 0, 0, 0, 0),
 ('No', 'Travel_Frequently', '25 - 34', 'Current Employees', 'R&D', 'Life Sciences', 'STAFF-5', 5, 'Female', 'Research Scientist', 'Married', 'Yes', 'Y', 3, 33, 1, 1392, 3, "Master's Degree", 1, 4, 56, 3, 1, 3, 2909, 23159, 1, 11, 3, 3, 80, 0, 8, 3, 8, 7, 3, 0),
 ('No', 'Travel_Rarely', '25 - 34', 'Current Employees', 'R&D', 'Medical', 'STAFF-7', 7, 'Male', 'Laboratory Technician', 'Married', 'No', 'Y', 3, 27, 1, 591, 2, 'High School', 1, 1, 40, 3, 1, 2, 3468, 16632, 9, 12, 3, 4, 80, 1, 6, 3, 2, 2, 2, 2),
 ('No', 'Travel_Frequently', '25 - 34', 'Current Employees', 'R&D', 'Life Sciences', 'STAFF-8', 8, 'Male', 'Laboratory Technician', 'Single', 'No', 'Y', 2, 32, 1, 1005, 2, 'Associates Degree', 1, 4, 79, 3, 1, 4, 3068, 11864, 0, 13, 3, 3, 80, 0, 8, 2, 7, 7, 3, 6),
 ('No', 'Travel_Rarely', 'Over 55', 'Current Employees', 'R&D', 'Medical', 'STAFF-10', 10, 'Female', 'Laboratory Technician', 'Married', 'Yes', 'Y', 3, 59, 1, 1324, 3, "Bachelor's Degree", 1, 3, 81, 4, 1, 1, 2670, 9964, 4, 20, 4, 1, 80, 3, 12, 2, 1, 0, 0, 0),
 ('No', 'Travel_Rarely', '25 - 34', 'Current Employees', 'R&D', 'Life Sciences', 'STAFF-11', 11, 'Male', 'Laboratory Technician', 'Divorced', 'No', 'Y', 2, 30, 1, 1358, 24, 'High School', 1, 4, 67, 3, 1, 3, 2693, 13335, 1, 22, 4, 2, 80, 1, 1, 3, 1, 0, 0, 0),
 ('No', 'Travel_Frequently', '35 - 44', 'Current Employees', 'R&D', 'Life Sciences', 'STAFF-12', 12, 'Male', 'Manufacturing Director', 'Single', 'No', 'Y', 2, 38, 1, 216, 23, "Bachelor's Degree", 1, 4, 44, 2, 3, 3, 9526, 8787, 0, 21, 4, 2, 80, 0, 10, 3, 9, 7, 1, 8),
 ('No', 'Travel_Rarely', '35 - 44', 'Current Employees', 'R&D', 'Medical', 'STAFF-13', 13, 'Male', 'Healthcare Representative', 'Married', 'No', 'Y', 3, 36, 1, 1299, 27, "Bachelor's Degree", 1, 3, 94, 3, 2, 3, 5237, 16577, 6, 13, 3, 2, 80, 2, 17, 2, 7, 7, 7, 7)]

We start by cleaning and extracting the required data from the dataset.  
We drop the columns that are not needed for the same.

In [9]:

%%sql

ALTER TABLE hr_data
drop column emp_no,
drop column employee_number,
drop column over_time,
drop column over_eighteen ,
drop column training_time ,
drop column daily_wages,
drop column employee,
drop column hourly_wages,
drop column job_level,
drop column monthly_income,
drop column monthly_wages,
drop column no_of_companies_worked,
drop column percent_salary_hike,
drop column standard_hours,
drop column stock_option_level, 
drop column total_working_years,
drop column years_at_company,
drop column years_in_current_role,
drop column years_since_last_promotion,
drop column performance_rating,
drop column years_with_current_manager;

 * postgresql://postgres:***@localhost/hr_attrition
Done.


[]

In the next step, to organize and arrange the data to make it easier to read and analyze, we create a view on our table by rearranging and renaming the columns.  

We also create a new column called **emp_no** the uniquely identifies each record in the table.

In [10]:
%%sql

create view hr_data_vw as 
select 10000 + row_number() over() as emp_no,
gender, marital_status, age_group, age, department,
education_field, education, job_role, business_travel, 
attrition, attrition_label, environment_satisfaction, 
job_satisfaction, job_involvement, relationship_satisfaction, work_life as work_life_balance,
distance_from_home, current_employee as active_employee
from hr_data;

 * postgresql://postgres:***@localhost/hr_attrition
1470 rows affected.


[]

In [12]:
%%sql

select * from hr_data_vw limit 10;

 * postgresql://postgres:***@localhost/hr_attrition
10 rows affected.


[(10001, 'Female', 'Single', '35 - 44', 41, 'Sales', 'Life Sciences', 'Associates Degree', 'Sales Executive', 'Travel_Rarely', 'Yes', 'Ex-Employees', 2, 4, 3, 1, 1, 1, 0),
 (10002, 'Male', 'Married', '45 - 54', 49, 'R&D', 'Life Sciences', 'High School', 'Research Scientist', 'Travel_Frequently', 'No', 'Current Employees', 3, 2, 2, 4, 3, 8, 1),
 (10003, 'Male', 'Single', '35 - 44', 37, 'R&D', 'Other', 'Associates Degree', 'Laboratory Technician', 'Travel_Rarely', 'Yes', 'Ex-Employees', 4, 3, 2, 2, 3, 2, 0),
 (10004, 'Female', 'Married', '25 - 34', 33, 'R&D', 'Life Sciences', "Master's Degree", 'Research Scientist', 'Travel_Frequently', 'No', 'Current Employees', 4, 3, 3, 3, 3, 3, 1),
 (10005, 'Male', 'Married', '25 - 34', 27, 'R&D', 'Medical', 'High School', 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 1, 2, 3, 4, 3, 2, 1),
 (10006, 'Male', 'Single', '25 - 34', 32, 'R&D', 'Life Sciences', 'Associates Degree', 'Laboratory Technician', 'Travel_Frequently', 'No', 'Current Employees', 4, 4, 3, 3, 2, 2, 1),
 (10007, 'Female', 'Married', 'Over 55', 59, 'R&D', 'Medical', "Bachelor's Degree", 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 3, 1, 4, 1, 2, 3, 1),
 (10008, 'Male', 'Divorced', '25 - 34', 30, 'R&D', 'Life Sciences', 'High School', 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 4, 3, 3, 2, 3, 24, 1),
 (10009, 'Male', 'Single', '35 - 44', 38, 'R&D', 'Life Sciences', "Bachelor's Degree", 'Manufacturing Director', 'Travel_Frequently', 'No', 'Current Employees', 4, 3, 2, 2, 3, 23, 1),
 (10010, 'Male', 'Married', '35 - 44', 36, 'R&D', 'Medical', "Bachelor's Degree", 'Healthcare Representative', 'Travel_Rarely', 'No', 'Current Employees', 3, 3, 3, 2, 2, 27, 1)]

In the next step we create a new table using the view.

In [13]:
%%sql

select * into hr_data1 from hr_data_vw;

 * postgresql://postgres:***@localhost/hr_attrition
1470 rows affected.


[]

We then drop our original table to use the old table's name for renaming the new table.

In [14]:
%%sql

drop table if exists hr_data cascade;

 * postgresql://postgres:***@localhost/hr_attrition
Done.


[]

In [15]:
%%sql

alter table hr_data1
rename to hr_data

 * postgresql://postgres:***@localhost/hr_attrition
Done.


[]

We check if our changes have been applied successfully before proceeding to the next steps.

In [16]:
%%sql

select * from hr_data limit 10;

 * postgresql://postgres:***@localhost/hr_attrition
10 rows affected.


[(10001, 'Female', 'Single', '35 - 44', 41, 'Sales', 'Life Sciences', 'Associates Degree', 'Sales Executive', 'Travel_Rarely', 'Yes', 'Ex-Employees', 2, 4, 3, 1, 1, 1, 0),
 (10002, 'Male', 'Married', '45 - 54', 49, 'R&D', 'Life Sciences', 'High School', 'Research Scientist', 'Travel_Frequently', 'No', 'Current Employees', 3, 2, 2, 4, 3, 8, 1),
 (10003, 'Male', 'Single', '35 - 44', 37, 'R&D', 'Other', 'Associates Degree', 'Laboratory Technician', 'Travel_Rarely', 'Yes', 'Ex-Employees', 4, 3, 2, 2, 3, 2, 0),
 (10004, 'Female', 'Married', '25 - 34', 33, 'R&D', 'Life Sciences', "Master's Degree", 'Research Scientist', 'Travel_Frequently', 'No', 'Current Employees', 4, 3, 3, 3, 3, 3, 1),
 (10005, 'Male', 'Married', '25 - 34', 27, 'R&D', 'Medical', 'High School', 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 1, 2, 3, 4, 3, 2, 1),
 (10006, 'Male', 'Single', '25 - 34', 32, 'R&D', 'Life Sciences', 'Associates Degree', 'Laboratory Technician', 'Travel_Frequently', 'No', 'Current Employees', 4, 4, 3, 3, 2, 2, 1),
 (10007, 'Female', 'Married', 'Over 55', 59, 'R&D', 'Medical', "Bachelor's Degree", 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 3, 1, 4, 1, 2, 3, 1),
 (10008, 'Male', 'Divorced', '25 - 34', 30, 'R&D', 'Life Sciences', 'High School', 'Laboratory Technician', 'Travel_Rarely', 'No', 'Current Employees', 4, 3, 3, 2, 3, 24, 1),
 (10009, 'Male', 'Single', '35 - 44', 38, 'R&D', 'Life Sciences', "Bachelor's Degree", 'Manufacturing Director', 'Travel_Frequently', 'No', 'Current Employees', 4, 3, 2, 2, 3, 23, 1),
 (10010, 'Male', 'Married', '35 - 44', 36, 'R&D', 'Medical', "Bachelor's Degree", 'Healthcare Representative', 'Travel_Rarely', 'No', 'Current Employees', 3, 3, 3, 2, 2, 27, 1)]

##### **In the next step, we start with the analyze phase**

##### Number of employees

In [17]:
%%sql

select count(emp_no) as No_of_Employees
from hr_data;

 * postgresql://postgres:***@localhost/hr_attrition
1 rows affected.


[(1470,)]

The organization consisted of a total of 1470 employees (those who have left + those who are still active)

##### Attrition Count

In [18]:
%%sql

select count(attrition) as Total_attrition_count
from hr_data
where attrition = 'Yes';

 * postgresql://postgres:***@localhost/hr_attrition
1 rows affected.


[(237,)]

Out of 1470 employees, 237 employees have left the organization.

##### Number of active employees (Employees still working at the organization)

In [20]:
%%sql

select
count(emp_no) - sum(case when attrition = 'Yes' then 1 else 0 end) as Active_Employees
from hr_data;

 * postgresql://postgres:***@localhost/hr_attrition
1 rows affected.


[(1233,)]

There are a total of 1233 active employees.

In [ ]:
# ALTERNATE QUERY:		select count(attrition) as Active_employees
#  						from hr_data
#  						where attrition = 'No';

##### Attrition Rate

In [21]:
%%sql

select concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from hr_data;

 * postgresql://postgres:***@localhost/hr_attrition
1 rows affected.


[('16.12%',)]

In [ ]:
# ALTERNATE QUERY: 	select cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text)||'&' as Attrition_rate
#  					from hr_data;

Thus 16.12% of employees have left the organization.

##### Average Employee Age

In [22]:
%%sql

select ceil(avg(Age)) as average_age
from hr_data;

 * postgresql://postgres:***@localhost/hr_attrition
1 rows affected.


[(Decimal('37'),)]

The average age of employees in the organization is 37.

##### Attrition by department

In [58]:
%%sql

select department,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from hr_data
group by department
order by 4 desc;

 * postgresql://postgres:***@localhost/hr_attrition
3 rows affected.


[('Sales', 92, 354, '20.63%'),
 ('HR', 12, 51, '19.05%'),
 ('R&D', 133, 828, '13.84%')]

It can be seen that even though the highest number of employees who have left the organization belong to the R&D department, the sales and HR departments have a higher attrition rate.

##### Attrition by Job role

In [24]:
%%sql

select job_role,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees
from hr_data
group by job_role
order by 2 desc;

 * postgresql://postgres:***@localhost/hr_attrition
9 rows affected.


[('Laboratory Technician', 62, 197),
 ('Sales Executive', 57, 269),
 ('Research Scientist', 47, 245),
 ('Sales Representative', 33, 50),
 ('Human Resources', 12, 40),
 ('Manufacturing Director', 10, 135),
 ('Healthcare Representative', 9, 122),
 ('Manager', 5, 97),
 ('Research Director', 2, 78)]

The **Laboratory Technician** position within the organization is the one which contributes highest to the attrition count.

##### Attrition by Gender

In [37]:
%%sql

select gender,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from hr_data
group by gender
order by 2 desc;

 * postgresql://postgres:***@localhost/hr_attrition
2 rows affected.


[('Male', 150, 732, '17.01%'), ('Female', 87, 501, '14.80%')]

The male attrition count and rate is higher than the female attrition count and rates. However, upon considering their distribution within the organization, they are almost similar.

##### Attrition by Age Group

In [28]:
%%sql

select age_group,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees
from hr_data
group by age_group
order by 2 desc;

 * postgresql://postgres:***@localhost/hr_attrition
5 rows affected.


[('25 - 34', 112, 442),
 ('35 - 44', 51, 454),
 ('Under 25', 38, 59),
 ('45 - 54', 25, 220),
 ('Over 55', 11, 58)]

It can be clearly seen that employees within the age group of 25-34 have contributed the highest to the attrition count.

##### Attrition by Education

In [29]:
%%sql

select education,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees
from hr_data
group by education
order by 2 desc;

 * postgresql://postgres:***@localhost/hr_attrition
5 rows affected.


[("Bachelor's Degree", 99, 473),
 ("Master's Degree", 58, 340),
 ('Associates Degree', 44, 238),
 ('High School', 31, 139),
 ('Doctoral Degree', 5, 43)]

##### Attrition by Work Distance from home (in kms)

In [30]:
%%sql


select x.work_distance, sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count
from (select distance_from_home, 
case when distance_from_home <= 8 then 'Near-by'
when distance_from_home>8 and distance_from_home<=16 then 'Far'
else 'Very far' end as work_distance
from hr_data
group by 1
order by 1) x
join hr_data y
on x.distance_from_home = y.distance_from_home
group by 1;

 * postgresql://postgres:***@localhost/hr_attrition
3 rows affected.


[('Near-by', 115), ('Far', 61), ('Very far', 61)]

No concrete trend can be identified for the contribution of travel distances to attrition rates as more employees who live near by have left the organization compared to the employees who live far away.

##### Attrition by Business travels

In [31]:
%%sql

select business_travel,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count
from hr_data
group by business_travel
order by 2 desc;

 * postgresql://postgres:***@localhost/hr_attrition
3 rows affected.


[('Travel_Rarely', 156), ('Travel_Frequently', 69), ('Non-Travel', 12)]

A higher number of employees who travel rarely have left the organization compared to those who travel frequently.

##### **Survery Score**

##### We look at how environment satisfaction affected the attrition of employees.

In [47]:
%%sql
select x.env_satisfaction as environment_satisfaction, sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from (select environment_satisfaction, 
case when environment_satisfaction = 1 then 'Very Unsatisfied'
when environment_satisfaction=2 then 'Not Satisfied'
when environment_satisfaction=3 then 'Satisfied'
when environment_satisfaction=4 then 'Very Satisfied'
else 'Other' end as env_satisfaction
from hr_data
group by 1
order by 1) x
join hr_data y
on x.environment_satisfaction = y.environment_satisfaction
group by 1
order by 4 desc;

 * postgresql://postgres:***@localhost/hr_attrition
4 rows affected.


[('Very Unsatisfied', 72, 212, '25.35%'),
 ('Not Satisfied', 43, 244, '14.98%'),
 ('Satisfied', 62, 391, '13.69%'),
 ('Very Satisfied', 60, 386, '13.45%')]

Here the environment satisfaction based upon the score is described as-  
1 - Very Unsatisfied  
2 - Not Satisfied  
3 - Satisfied  
4 - Very Satisfied  

It is clear that an the lower the environment satisfaction score, the higher the attrition.  

Note: Here the attrition rate is row wise i.e. for each category

##### We take a similar approach to find out how job satisfaction affected the attrition of employees.

In [49]:
%%sql
select x.job_satisfaction_described as job_satisfaction, sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from (select job_satisfaction, 
case when job_satisfaction = 1 then 'Very Unsatisfied'
when job_satisfaction=2 then 'Not Satisfied'
when job_satisfaction=3 then 'Satisfied'
when job_satisfaction=4 then 'Very Satisfied'
else 'Other' end as job_satisfaction_described
from hr_data
group by 1
order by 1) x
join hr_data y
on x.job_satisfaction = y.job_satisfaction
group by 1
order by 4 desc;

 * postgresql://postgres:***@localhost/hr_attrition
4 rows affected.


[('Very Unsatisfied', 66, 223, '22.84%'),
 ('Satisfied', 73, 369, '16.52%'),
 ('Not Satisfied', 46, 234, '16.43%'),
 ('Very Satisfied', 52, 407, '11.33%')]

Here job satisfaction based upon the score is described similar to environment satisfaction above -  
1 - Very Unsatisfied  
2 - Not Satisfied  
3 - Satisfied  
4 - Very Satisfied  

It is clear that an the lower the job satisfaction score, higher the attrition.  

Note: Here the attrition rate is row wise i.e. for each category

##### Next, we take a look at how employee job involvement affects attrition

In [55]:
%%sql
select x.job_involvement_described as job_involvement, sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from (select job_involvement, 
case when job_involvement = 1 then 'Very less'
when job_involvement=2 then 'Less'
when job_involvement=3 then 'High'
when job_involvement=4 then 'Very High'
else 'Other' end as job_involvement_described
from hr_data
group by 1
order by 1) x
join hr_data y
on x.job_involvement = y.job_involvement
group by 1
order by 4;

 * postgresql://postgres:***@localhost/hr_attrition
4 rows affected.


[('High', 125, 743, '14.40%'),
 ('Less', 71, 304, '18.93%'),
 ('Very less', 28, 55, '33.73%'),
 ('Very High', 13, 131, '9.03%')]

Here job involvement based upon the score is described as -  
1 - Very Less involvement  
2 - Less involvement  
3 - High involvement  
4 - Very High involvement  

It is clear that the lower the employee involvement, higher is the attrition.  

Note: Here the attrition rate is row wise i.e. for each category

##### Finally, we find out how work-life balance affects the attrition rate.

In [56]:
%%sql

select work_life_balance as work_life_balance_score,
sum(case when attrition = 'Yes' then 1 else 0 end) as Attrition_count,
sum(case when attrition = 'No' then 1 else 0 end) as Active_employees,
concat(cast(round(sum(case when attrition = 'Yes' then 1 else 0 end)::decimal/count(emp_no)*100,2) as text),'%') as Attrition_rate
from hr_data
group by work_life_balance
order by 1;

 * postgresql://postgres:***@localhost/hr_attrition
4 rows affected.


[(1, 25, 55, '31.25%'),
 (2, 58, 286, '16.86%'),
 (3, 127, 766, '14.22%'),
 (4, 27, 126, '17.65%')]

Thus, people who are unable to maintain a poor work life balance are more likely to leave the organization.

##### Thus, employees who are least involved in their job, who have poor work life balances and those who are very unsatisfied with their jobs and work environments are more likely to leave the organization. More data is need to determine the reasons for dissatisfaction and poor involvement to perform further analysis.

### Note: I have created a dashboard in tableau to view these findings at a glance and filter them out based on different data points.  
### Click here to view the Dashboard: [HR Attrition Analysis dashboard](https://public.tableau.com/app/profile/jayesh25/viz/HRAttritionAnalysis_16866548755520/Dashboard1)